migrations/sqls/20190725182023-cancel-interaction-rpc-0184-up.sql

```diff
@@ -0,0 +1,18 @@
+/*  db-migrate create cancel-interaction-rpc-0184 -e pg-staging */
+
+INSERT INTO function_group_hmi_levels(function_group_id, permission_name, hmi_level)
+(SELECT id AS function_group_id, 'CancelInteraction' AS permission_name, 'FULL'::hmi_level AS hmi_level
```

Comment:
Recommend using this cleaner SELECT statement for the INSERT. Give it a try?

```suggestion:+14
SELECT id AS function_group_id, 'CancelInteraction' AS permission_name, v.hmi_level AS hmi_level
FROM function_group_info
CROSS JOIN (
    VALUES ('FULL'), ('LIMITED'), ('BACKGROUND')
) AS v(hmi_level)
WHERE property_name = 'Base-4'
AND NOT EXISTS(
    SELECT 1 
    FROM function_group_hmi_levels 
    WHERE property_name = 'Base-4' 
    AND permission_name = 'CancelInteraction'
);
```

Reply:
This works with one minor change and looks a lot cleaner. I changed the selected v.hmi_level to v.hmi_level::hmi_level to cast this as a hmi_level enum before inserting.